### PR TITLE
Add nil check to server group hint creation

### PR DESCRIPTION
**Pull request by jhg03a**
I ran into a provider panic while trying to unassign a server group from a server.  By deferring to the default behavior, the nil check is sufficient to effectively unassign a server group.

### CODE REVIEW

**Comment by jhg03a** [comment]
The test failures appear to be unrelated to the change itself but rather constructing the devstack.

File: openstack/resource_openstack_compute_instance_v2.go [hunk 1]
```diff
@@ -1403,31 +1403,31 @@ func getFlavorID(computeClient *gophercloud.ServiceClient, d *schema.ResourceDat
...
-
-	if m["group"] != nil {
-		buf.WriteString(fmt.Sprintf("%s-", m["group"].(string)))
-	}
-
-	if m["target_cell"] != nil {
-		buf.WriteString(fmt.Sprintf("%s-", m["target_cell"].(string)))
-	}
+	if v != nil && v.(map[string]interface{}) != nil {
+		m := v.(map[string]interface{})
```
Comment (ozerovandrei) [inline openstack/resource_openstack_compute_instance_v2.go:1407]: I guess it's better not to nest all the logic under a single `if {}` and we can also cast just a single time like this:
```go
m, ok := v.(map[string]interface{})
if !ok {
  return hashcode.String(buf.String())
}
if m == nil {
  return hashcode.String(buf.String())
}

// other stuff

return hashcode.String(buf.String())
```

**Review by ozerovandrei — CHANGES_REQUESTED** [review]

File: openstack/resource_openstack_compute_instance_v2.go (hunk 1 above)
Comment (jhg03a) [reply openstack/resource_openstack_compute_instance_v2.go]: Addressed with commit

**Comment by jhg03a** [comment]
The CI failures appear to be unrelated to the change not being able to find a golang compiler.

**Comment by ozerovandrei** [comment]
@jhg03a could you please also resolve `golangci-lint` issue?
I can merge this without `openlab/check` since those changes looks safe.

**Review by ozerovandrei — APPROVED** [review]
LGTM but we need tests to be fixed first #1253

**Comment by jhg03a** [comment]
Those acceptance test failures are not related to the PR, but rather https://github.com/terraform-provider-openstack/terraform-provider-openstack/issues/1253

**Comment by nikParasyr** [comment]
recheck

**Review by ozerovandrei — APPROVED** [review]

**Pull request merged**